stock_available_to_promise_release_carrier_alternative/models/stock_picking.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _carrier_can_be_changed(self):
+        self.ensure_one()
+        active_moves = self.move_ids.filtered(
+            lambda move: move.state not in ("done", "cancel")
+        )
+        if active_moves:
+            for move in active_moves:
+                if not move.need_release:
+                    return False
+            return True
+
+    def check_alternative_carriers(self):
+        if not self._carrier_can_be_changed():
+            return
+        carrier = self.get_preferred_carrier()
+        if not carrier:
+            return {
+                "warning": {
```

Comment:
I don't see a reason for this method to be called on the picking without releasing it. Otherwise, it's missing important steps currently located on stock.move

Reply:
drop this warning